diff.c

```diff
@@ -4595,7 +4598,8 @@ void repo_diff_setup(struct repository *r, struct diff_options *options)
 
 	options->color_moved = diff_color_moved_default;
 	options->color_moved_ws_handling = diff_color_moved_ws_default;
-
+	if(!diff_ignore_submodule_config)
+		handle_ignore_submodules_arg(options, "untracked");
```

Comment:
The common strategy is to set the default _first_, and let the config override it. So it would make more sense to append a line after `options->orderfile = diff_order_file_cfg`: `options->flags.ignore_untracked_in_submodules = 1;`.